site/_posts/2017-05-23-0.4.0-release.md

```diff
@@ -0,0 +1,111 @@
+implementation develops, we will eventually want to include JS in the
+integration test suite along with Java and C++ to ensure wire
+cross-compatibility.
+
+### Python Support for Apache Parquet on Windows
+
+With the [1.1.0 C++ release][7] of [Apache Parquet][8], we have enabled the
+`pyarrow.parquet` extension on Windows for Python 3.5 and 3.6. This should
+appear in conda-forge packages and PyPI in the near future. Developers can
+follow the [source build instructions][9].
+
+### Generalizing Arrow Streams
+
+In the 0.2.0 release, we defined the first version of the Arrow streaming
+binary format for low-cost messaging with columnar data. These streams presume
+that the message components are written as a continuous byte stream over a
+socket or file.
+
+We would like to be able to support other other transport protocols, like
+[gRPC][3], for the message components of Arrow streams. To that end, in C++ we
+defined an abstract stream reader interface, for which the current contiguous
+streaming format is one implementation:
+
+{% highlight cpp %}
+class RecordBatchReader {
+ public:
+  virtual std::shared_ptr<Schema> schema() const = 0;
+  virtual Status GetNextRecordBatch(std::shared_ptr<RecordBatch>* batch) = 0;
+};
+{% endhighlight %}
+
+It would also be good to define abstract stream reader and writer interfaces in
+the Java implementation.
+
+In an upcoming blog post, we will explain in more depth how Arrow streams work,
+but you can learn more about them by reading the [IPC specification][4].
+
+### C++ and Cython API for Python Extensions
+
+As other Python libraries with C or C++ extensions use Apache Arrow, they will
+need to be able to return Python objects wrapping the underlying C++
+objects. In this release, we have implemented a prototype C++ API which enables
+Python wrapper objects to be constructed from C++ extension code:
+
+{% highlight cpp %}
+#include "arrow/python/pyarrow.h"
+
+if (!arrow::py::import_pyarrow()) {
+  // Error
+}
+
+std::shared_ptr<arrow::RecordBatch> cpp_batch = GetData(...);
+PyObject* py_batch = arrow::py::wrap_batch(cpp_batch);
+{% endhighlight %}
+
+This API is intended to be usable from Cython code as well:
+
+{% highlight cython %}
+cimport pyarrow
+pyarrow.import_pyarrow()
+{% endhighlight %}
+
+### Python Wheel Installers on macOS
+
+With this release, `pip install pyarrow` works on macOS (OS X) as well as
+Linux. We are working on providing binary wheel installers for Windows as well.
+
+[1]: https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20status%20in%20(Resolved%2C%20Closed)%20AND%20fixVersion%20%3D%200.4.0
+[2]: http://arrow.apache.org/install
+[3]: http://grpc.io/
+[4]: http://arrow.apache.org/docs/ipc.html
+[6]: http://arrow.apache.org/install
+[7]: https://github.com/apache/parquet-cpp/releases/tag/apache-parquet-cpp-1.1.0
+[8]: http://parquet.apache.org
+[9]: https://github.com/apache/arrow/blob/master/python/doc/source/development.rst
```

Comment:
Shouldn't this link to the rendered version at https://arrow.apache.org/docs/python/development.html?